code/modules/mob/living/carbon/human/human.dm

```diff
@@ -633,6 +633,10 @@
 		//If you dragged them to you and you're aggressively grabbing try to fireman carry them
 		INVOKE_ASYNC(src, PROC_REF(fireman_carry), grabbed)
 		return COMSIG_GRAB_SUCCESSFUL_SELF_ATTACK
+	if(isxenorunner(pulling))
+		if(grabbed.stat == CONSCIOUS && stat == CONSCIOUS)
+			INVOKE_ASYNC(grabbed, TYPE_PROC_REF(/mob/living/carbon/xenomorph/runner, carry_human), src, TRUE)
+			return COMSIG_GRAB_SUCCESSFUL_SELF_ATTACK
```

Comment:
why is this snowflaked to the runner mob instead of a component?

Reply:
Ive never worked with comps at all; If you wanna like help me with that i can turn it into that.

Reply:
Actually don't we already have a riding component of some sort?

Reply:
> Actually don't we already have a riding component of some sort?

We do, this uses that; This is basically copied from how crusher does it.